### PR TITLE
[ENG-6368] Share summary graphs

### PR DESCRIPTION
**Pull request by futa-ikeda**
-   Ticket: [ENG-6368]
-   Feature flag: n/a

## Purpose
- Hook up graphs for licenses, addon usage and storage locations to actual data

## Summary of Changes
- Add loading/error state to chart component
- Add task to fetch data from SHARE to chart-wrapper component

## Screenshot(s)
- Finished graphs:
image

- Loading/error states:
image

## QA Notes

[ENG-6368]: https://openscience.atlassian.net/browse/ENG-6368?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by brianjgeiger — APPROVED** [review]

File: app/institutions/dashboard/-components/chart-kpi-wrapper/component.ts [hunk 1]
```diff
@@ -185,91 +199,47 @@ export default class ChartKpiWrapperComponent extends Component<TotalCountChartW
...
-                {
-                    label: metric.name,
-                    total: metric.numberOfUsers,
-                } as ChartDataModel,
-            );
-        });
-        return storageRegionsData;
-    }
+        };
+        this.kpiCharts[index] = kpiChartObject;
```
Comment (aaxelb) [inline app/institutions/dashboard/-components/chart-kpi-wrapper/component.ts:241]: this is getting a bit convoluted and brittle, overwriting an object in an array by constant index (especially when `title` and `chartType` are unchanged but computed two ways) and then re-rendering everything that depends on that array (multiple times, it seems?)

i gather you don't want to make the whole set of charts block on these value-search requests, which makes sense... might simplify things if the `ChartKpiWrapper::ChartKpi` component understood the task instance you're passing it and automatically used `this.args.data.taskInstance.value` (when `this.args.data.chartData` is missing) -- then this task could have just one argument (`propertyPath`), return its `chartData`, and not have to do anything with `this.kpiCharts`

**Review by aaxelb — COMMENTED** [review]
mainly a suggestion to simplify some confusing logic, but that confusing logic does seem like it'll probably work (tho may be brittle to updates) -- your call

File: app/institutions/dashboard/-components/chart-kpi-wrapper/component.ts (hunk 1 above)
Comment (futa-ikeda) [reply app/institutions/dashboard/-components/chart-kpi-wrapper/component.ts:241]: Ahh, this is a much more sensible way to implement this. Will rework these components now

**Review by aaxelb — COMMENTED** [review]
couple nits for clarity and a type annotation correction

**Review by aaxelb — APPROVED** [review]
🎐

**Pull request merged**